storage/badger/results.go

```diff
@@ -21,7 +23,7 @@ func NewExecutionResults(db *badger.DB) *ExecutionResults {
 }
 
 func (r *ExecutionResults) store(result *flow.ExecutionResult) func(*badger.Txn) error {
-	return operation.InsertExecutionResult(result)
+	return operation.SkipDuplicates(operation.InsertExecutionResult(result))
```

Comment:
I believe we've said that we can skip duplicates because if the results are different, they're going to have different IDs anyways, is that right?

Reply:
yes, we can store different results, but when we attempt to index the second result which is different, then the stored one, it will fail.

A nice thing to have in the future is that: The indexing and storing could actually be implemented within one function, and within one transaction. Such that we won't store a second result if it can't be indexed. We could do this along with [making saving results in one transaction](https://github.com/dapperlabs/flow-go/issues/4967)